network/src/peer_registry.rs

```diff
@@ -218,20 +228,17 @@ impl PeerRegistry {
 
     /// Get session id by peer id
     pub fn get_key_by_peer_id(&self, peer_id: &PeerId) -> Option<SessionId> {
-        self.peers.values().find_map(|peer| {
-            if &peer.peer_id == peer_id {
-                Some(peer.session_id)
-            } else {
-                None
-            }
+        self.peers.iter().find_map(|(session_id, peer)| {
+            extract_peer_id(&peer.connected_addr).and_then(|pid| {
```

Comment:
A peer must have a peer_id, so we can using `expect` here, or add a peer_id method to the Peer structure.

Reply:
it's just a different style of code, we can use expect here, but I prefer chain invoking here